src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,238 @@
+        """
+        if isinstance(params, Namespace):
+            params = vars(params)
+
+        self.experiment.save_hyperparameters(params)
+
+    @override
+    def after_save_checkpoint(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Lightning Logger function. Handles checkpoint saving to the logger after
+        the ModelCheckpoint Callback of the Lightning Trainer is called.
+        The checkpoints are logged as artifacts.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance to manage checkpointing.
+        """
+        if self._log_model == "all" or (
+            self._log_model is True and checkpoint_callback.save_top_k == -1
+        ):
+            self._scan_and_log_checkpoints(checkpoint_callback)
+        elif self._log_model is True:
+            self._checkpoint_callback = checkpoint_callback
+
+    def _scan_and_log_checkpoints(
+        self,
+        checkpoint_callback: ModelCheckpoint
+    ) -> None:
+        """Scans and logs checkpoints as artifacts in the experiment.
+
+        This function retrieves new checkpoints, logs them as artifacts, and saves
+        related metadata, including checkpoint score, filename, and any model checkpoint
+        parameters.
+
+        Args:
+            checkpoint_callback (ModelCheckpoint): Callback instance used for checkpointing.
+        """
+        # Get checkpoints to be saved with associated score
+        checkpoints = _scan_checkpoints(checkpoint_callback, self._logged_model_time)
+
+        # Log iteratively all new checkpoints
+        for t, p, s, tag in checkpoints:
+            metadata = {
+                "score": s.item() if isinstance(s, Tensor) else s,
+                "original_filename": Path(p).name,
+                checkpoint_callback.__class__.__name__: {
+                    k: getattr(checkpoint_callback, k)
+                    for k in [
+                        "monitor",
+                        "mode",
+                        "save_last",
+                        "save_top_k",
+                        "save_weights_only",
+                        "_every_n_train_steps",
+                    ]
+                    # ensure it does not break if `ModelCheckpoint` args change
+                    if hasattr(checkpoint_callback, k)
+                },
+            }
+            if p == checkpoint_callback.best_model_path:
+                aliases = ["latest", "best"]
+            else:
+                aliases = ["latest"]
+            artifact_path = Path(p).stem
+
+            # Log the checkpoint
+            self.experiment.log(
+                item=p,
+                identifier=artifact_path,
+                kind='artifact'
+            )
+
+            with tempfile.TemporaryDirectory(
+                prefix="test",
+                suffix="test",
+                dir=os.getcwd()
+            ) as tmp_dir:
```

Comment:
In this case it would probably be nicer to do 
```python
with ...(
   prefix, suffix, dir
) as tmp_dir:
```
i.e. that you have the middle ones on a single line.